### ISSUE TITLE
False positives with `unhandled_throwing_task`

### ISSUE THREAD

**Issue by noremac** [realm/SwiftLint#4984]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Here's a minimal code snippet that incorrectly triggers the newly introduced `unhandled_throwing_task` rule.

```swift
let result = await Task {
  throw CancellationError()
}.result
```

The Task is a throwing one, but I am calling the non-throwing `result` function on it so I believe the error is incorrect. (Just in case it's interesting: I'm doing this so that I can do `XCTAssertNotNil(result.failure)` in some tests until `XCTAssertNoThrow` works better with async stuff 🤷🏻‍♀️.)

### Environment

* SwiftLint version: 0.52.0
* Installation method used: downloading the pre-built binary from GitHub
* Paste your configuration file:

```yml
disabled_rules:
  - cyclomatic_complexity
  - file_length
  - for_where
  - force_cast
  - force_try
  - function_body_length
  - function_parameter_count
  - identifier_name
  - large_tuple
  - line_length
  - nesting
  - opening_brace
  - prohibited_interface_builder
  - single_test_class
  - todo
  - trailing_comma
  - type_body_length
  - type_name
  # - unhandled_throwing_task

opt_in_rules:
  - closure_end_indentation
  - closure_spacing
  - collection_alignment
  - comma_inheritance
  - conditional_returns_on_newline
  - contains_over_filter_count
  - contains_over_filter_is_empty
  - contains_over_first_not_nil
  - contains_over_range_nil_comparison
  - convenience_type
  - discouraged_assert
  - discouraged_none_name
  - discouraged_object_literal
  - empty_collection_literal
  - empty_count
  - empty_string
  - empty_xctest_method
  - fallthrough
  - fatal_error_message
  - first_where
  - flatmap_over_map_reduce
  - joined_default_parameter
  - last_where
  - legacy_random
  - lower_acl_than_parent
  # - missing_docs
  - modifier_order
  - multiline_literal_brackets
  - multiline_parameters
  - multiline_parameters_brackets
  - operator_usage_whitespace
  - overridden_super_call
  - prohibited_interface_builder
  - prohibited_super_call
  - reduce_boolean
  - reduce_into
  - redundant_type_annotation
  - self_binding
  - single_test_class
  - sorted_first_last
  - sorted_imports
  - static_operator
  - superfluous_disable_command
  - switch_case_on_newline
  - toggle_bool
  - unavailable_function
  - unneeded_parentheses_in_closure_argument
  - xct_specific_matcher

blanket_disable_command:
  allowed_rules:
    - missing_docs

excluded:
  - .build
  - Scripts
```

* Are you using nested configuration files: no
* Which Xcode version are you using: 14.3

**Comment by jpsim** [comment]
FYI @kylebshr

**Comment by kylebshr** [comment]
Oops, should be an easy fix - will address in a few minutes. Good catch!

**Comment by noremac** [comment]
Thank you!
